File: app/models/card.rb
class Card
  attr_reader :rank, :suit
  def initialize(rank:, suit:)
    @rank = rank
    @suit = suit
  end

  def value
    "#{@suit}#{@rank}"
  end

  def as_json
    {
      'rank' => @rank,
      'suit' => @suit
    }
  end

  def self.from_json(json)
    self.new(rank: json['rank'], suit: json['suit'])
  end
end

File: app/javascript/models/card.js
class Card {
  constructor(rank, suit) {
    this._rank = rank
    this._suit = suit
  }

  rank() {
    return this._rank
  }

  suit() {
    return this._suit
  }

  value() {
    return this._rank + this._suit
  }
}

export default Card
